src/DI/ConsoleExtension.php

```diff
@@ -39,19 +40,30 @@ class ConsoleExtension extends CompilerExtension
 		'lazy' => FALSE,
 	];
 
+	/** @var bool */
+	private $cliMode;
+
+	/**
+	 * @param bool $cliMode
+	 */
+	public function __construct($cliMode = FALSE)
```

Comment:
I'm quite confused by this parameter + the enabled config. Are you certain we need both? I can tell already that it would take me a few minutes to determine how things work. (Too lazy to do it now.)

Reply:
Well, I don't know how to solve this example: enable extension for cli/non-cli and does not override http URL in non-cli mode.

Reply:
Hmm. I agree that it must look little bit weird. Behavior for enabled is the same as (not) registering extension. Extension should probably be registered in CLI-only config file if you don't want console in HTTP mode.

Reply:
Anyone knows how this works in Symfony?

Reply:
And maybe $cliMode should be without default value to force user configure if it is really cli mode or not. Using just true/false could create weird behavior.

Reply:
I came up with something maybe crazy/magic, but it can works.

You can use initialize method in container like this:

```
	public function afterCompile(Nette\PhpGenerator\ClassType $class)
	{
		parent::afterCompile($class);

		$initialize = $class->methods['initialize'];
		$config = $this->validateConfig($this->defaults);

		if (isset($config['url'])) {
			$initialize->addBody(sprintf("
			if (!$this->hasService('http.request')) {
				$this->addService('http.request', new \Nette\Http\Request(%s));
			}
		", $config['url']));
		}
	}
```

Reply:
It is not clear from api how it works. But I have another idea. We can use afterCompile to overwrite http request only if PHP_SAPI === 'cli' at runtime.

Reply:
Well @pilec, it's a brave idea :-D

Reply:
@mabar PHP_SAPI is problematic, because it's useful when you can control by yourself when it's generated. You need some parameters for that.